### PR TITLE
Reserved Instance Net SDK public PR

### PR DESCRIPTION
**Pull request by corquiri**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by corquiri** [comment]
This is .Net SDK from Reserved Instance (=Reservations) that was generated by swagger in following PR:
https://github.com/Azure/azure-rest-api-specs/pull/1962
Tests are added as well.

File: src/SDKs/Reservations/Microsoft.Azure.Management.Reservations/Microsoft.Azure.Management.Reservations.csproj [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    <Description>Microsoft.Azure.Management.Reservations Library</Description>
+    <AssemblyName>Microsoft.Azure.Management.Reservations</AssemblyName>
+    <VersionPrefix>1.0.0-preview</VersionPrefix>
+  </PropertyGroup>
+  
+  <PropertyGroup>
+    <TargetFrameworks>net452;netstandard1.4</TargetFrameworks>
+  </PropertyGroup>
+  
+  <!--
```
Comment (shahabhijeet) [inline src/SDKs/Reservations/Microsoft.Azure.Management.Reservations/Microsoft.Azure.Management.Reservations.csproj]: @corquiri remove commented code

File: src/SDKs/Reservations/Microsoft.Azure.Management.Reservations/generate.cmd [hunk 2]
```diff
@@ -0,0 +1,7 @@
+::
+:: Microsoft Azure SDK for Net - Generate library code
+:: Copyright (C) Microsoft Corporation. All Rights Reserved.
+::
+
+@echo off
+call %~dp0..\..\..\..\tools\generate.cmd compute/resource-manager %*
```
Comment (shahabhijeet) [inline src/SDKs/Reservations/Microsoft.Azure.Management.Reservations/generate.cmd]: @corquiri the path should be according to your RP path in the spec repo

File: src/SDKs/Reservations/Reservations.Tests/Reservations.Tests.csproj [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    <PackageId>Reservations.Tests</PackageId>
+    <Description>Reservations.Tests Class Library</Description>
+    <Authors>Microsoft Corporation</Authors>
+    <AssemblyName>Reservations.Tests</AssemblyName>
+    <VersionPrefix>1.0.0-preview</VersionPrefix>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp1.1</TargetFramework>
+  </PropertyGroup>
+
+  <!--
```
Comment (shahabhijeet) [inline src/SDKs/Reservations/Reservations.Tests/Reservations.Tests.csproj]: @corquiri remove commented code.

File: src/SDKs/Reservations/Reservations.Tests/Reservations.Tests.csproj [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.5.0-preview-20170810-02" />
+    <PackageReference Include="Microsoft.Rest.ClientRuntime.Azure" Version="3.3.9" />
+    <PackageReference Include="Microsoft.Rest.ClientRuntime.Azure.TestFramework" Version="1.7.2" />
+    <PackageReference Include="xunit" Version="2.2.0" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.3.0-beta3-build3705" />
+  </ItemGroup>
+  -->
+
+  <ItemGroup>
+    <ProjectReference Include="..\Microsoft.Azure.Management.Reservations\Microsoft.Azure.Management.Reservations.csproj" />
```
Comment (shahabhijeet) [inline src/SDKs/Reservations/Reservations.Tests/Reservations.Tests.csproj]: @corquiri add reference to latest version of resource manager nuget package and record test using the latest version of resource manager

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
Missing meta data file (.txt), if you use generate.cmd to generate your SDK it will create the meta data file for your SDK.

File: src/SDKs/Reservations/Reservations.Tests/Reservations.Tests.csproj (hunk 4 above)
Comment (corquiri) [reply src/SDKs/Reservations/Reservations.Tests/Reservations.Tests.csproj]: capture

RM reference is already in Nuget dependency of Test project as shown

File: src/SDKs/Reservations/Reservations.Tests/Reservations.Tests.csproj (hunk 3 above)
Comment (corquiri) [reply src/SDKs/Reservations/Reservations.Tests/Reservations.Tests.csproj]: removed

File: src/SDKs/Reservations/Microsoft.Azure.Management.Reservations/generate.cmd (hunk 2 above)
Comment (corquiri) [reply src/SDKs/Reservations/Microsoft.Azure.Management.Reservations/generate.cmd]: corrected

File: src/SDKs/Reservations/Microsoft.Azure.Management.Reservations/Microsoft.Azure.Management.Reservations.csproj (hunk 1 above)
Comment (corquiri) [reply src/SDKs/Reservations/Microsoft.Azure.Management.Reservations/Microsoft.Azure.Management.Reservations.csproj]: removed

File: src/SDKs/Reservations/Reservations.Tests/Reservations.Tests.csproj [hunk 5]
```diff
@@ -0,0 +1,27 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.test.reference.props'))" />
+  <PropertyGroup>
+    <PackageId>Reservations.Tests</PackageId>
+    <Description>Reservations.Tests Class Library</Description>
+    <Authors>Microsoft Corporation</Authors>
+    <AssemblyName>Reservations.Tests</AssemblyName>
+    <VersionPrefix>1.0.0-preview</VersionPrefix>
+    <OutputType>Exe</OutputType>
```
Comment (shahabhijeet) [inline src/SDKs/Reservations/Reservations.Tests/Reservations.Tests.csproj]: @corquiri these tests are not an app/exe. If you model the test project looking at other test projects that will help to make it consistent across.

File: src/SDKs/Reservations/Reservations.Tests/Reservations.Tests.csproj [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    <TargetFramework>netcoreapp1.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\Microsoft.Azure.Management.Reservations\Microsoft.Azure.Management.Reservations.csproj" />
```
Comment (shahabhijeet) [inline src/SDKs/Reservations/Reservations.Tests/Reservations.Tests.csproj]: you are still missing ResourceManager 1.6.0 (latest version)
You are using the implicitly added reference to record all your tests. That version is 1.1.0 which is really old and was published last year. You need to use the latest version.

File: src/SDKs/Reservations/Microsoft.Azure.Management.Reservations/Microsoft.Azure.Management.Reservations.csproj [hunk 7]
```diff
@@ -0,0 +1,13 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
+  <PropertyGroup>
+    <PackageId>Microsoft.Azure.Management.Reservations</PackageId>
+    <Description>Microsoft.Azure.Management.Reservations Library</Description>
+    <AssemblyName>Microsoft.Azure.Management.Reservations</AssemblyName>
+    <VersionPrefix>1.0.0-preview</VersionPrefix>
```
Comment (shahabhijeet) [inline src/SDKs/Reservations/Microsoft.Azure.Management.Reservations/Microsoft.Azure.Management.Reservations.csproj]: @corquiri sorry forgot to mention this earlier in my review, but you are missing packageReleaseNotes take a look at this example
https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/BatchAI/Management.BatchAI/Microsoft.Azure.Management.BatchAI.csproj#L10
having package release notes helps your customers/users to know what is in the package (treat it as what you would like to tell your users what this nuget package has)

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/Reservations/Microsoft.Azure.Management.Reservations/Microsoft.Azure.Management.Reservations.csproj (hunk 7 above)
Comment (corquiri) [reply src/SDKs/Reservations/Microsoft.Azure.Management.Reservations/Microsoft.Azure.Management.Reservations.csproj]: Release note added

File: src/SDKs/Reservations/Reservations.Tests/Reservations.Tests.csproj (hunk 5 above)
Comment (corquiri) [reply src/SDKs/Reservations/Reservations.Tests/Reservations.Tests.csproj]: removed

File: src/SDKs/Reservations/Reservations.Tests/Reservations.Tests.csproj (hunk 6 above)
Comment (corquiri) [reply src/SDKs/Reservations/Reservations.Tests/Reservations.Tests.csproj]: latest version added

**Comment by corquiri** [comment]
Sent @shahabhijeet an email but leaving comment here just in case:

I was resolving your comment and found out that under SDKs\Reservations, I have two following folders:

Microsoft.Azure.Management.Reservations
Reservations.Tests

But looking at other modules it seems this should be 

Management.Reservations
Reservations.Tests

I took over this project from someone else so not sure why this is not following naming convention – I wonder if the renaming is required at this point. If so, should I create a new project and copy autorest-generated code in the newly created project?

**Comment by shahabhijeet** [comment]
@azuresdkci retest this please

File: src/SDKs/Reservations/Reservations.Tests/Reservations.Tests.csproj (hunk 6 above)
Comment (corquiri) [reply src/SDKs/Reservations/Reservations.Tests/Reservations.Tests.csproj]: I actually think this reference is not used in the project, I removed the reference and build succeeded and test all passed

**Comment by corquiri** [comment]
@shahabhijeet let me know if any more changes are needed, if not, would like to merge this ASAP. Thanks!

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**